clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -3254,7 +3263,10 @@ protected void validateMessage(
    * consumption in leader is ahead of drainer, leaders and drainers are processing messages at different paces.
    */
   protected void cloneProducerStates(int partition, KafkaDataIntegrityValidator validator) {
-    this.kafkaDataIntegrityValidator.cloneProducerStates(partition, validator);
+    this.kafkaDataIntegrityValidator
+        .cloneProducerStates(PartitionTracker.TopicType.VERSION_TOPIC, partition, validator);
+    this.kafkaDataIntegrityValidator
```

Comment:
I think we need to carefully think about the future phases for this project.
In the high-level, if we don't choose eDIV, but use a full snapshot, here is the producing logic in my mind:
1. Config a time-interval.
2. When the ingestion progress passes the time interval, `SIT` will produce a full snapshot to VersionTopic per partition.

There are a few prod fabircs and with current logic, RT topic in each prod fabric will be consumed independently, that means the RT DIV can be updated while `SIT` is generating a full snapshot per partition.
There is a race condition, and to avoid it, with today's architecture, we will need to maintain a full snapshot per source fabric per partition.
Please double check in case I had some misunderstanding.

Reply:
Gaojie, your understanding is correct and I think I see your concern here. Let me try if I understand it right,

The race condition could happen for AA store and when multi consumer thread updates and snapshots the same partition DIV, to address it, potentially we can:
 
1. as you suggested, divides our RT DIV further and into per colo RT DIV, so that per-source-fabric and per-partition DIV is updated by a single consumer thread. (no lock is required, but we may have per-source-fabric per-partition RT DIV HB messages) 
2. or introduce a `partition-level` wr-lock, so that any consumer threads working on same partition require to hold a w-lock when updating, and a r-lock when snapshotting. (sounds more performance overhead)

Reply:
2) carries a lot of overheads and it means the updates to the same partition from different RT topics will be executed sequentially, which is the opposite to the direction of parallelizing the processing for AA/WC workload.

Reply:
Makes sense. Working on a per-fabric RT DIV change.